### PR TITLE
Create and test PausableToken Contract

### PR DESCRIPTION
**Pull request by DavidKnott**
Creates PausableToken contract that inherits from the StandardTokenContract.
Change Pauable contract to use ```pause``` as opposed to ```emergency```
PausableToken has an owner who is able to:

Pause all token transfers
Unpause all token transfers

This contract was created in response to issue #194 
It's fully tested in PausableToken.js and all tests are passing.

### CODE REVIEW

**Comment by maraoz** [comment]
Amazing work @DavidKnott!!! Thanks a lot.
Do you mind refactoring the code so that it inherits from `Pausable`?
(https://github.com/OpenZeppelin/zeppelin-solidity/blob/master/contracts/lifecycle/Pausable.sol)

I actually like the modifier and event names better here, so if you can also modify `Pausable` to adopt this new names that would be great.
What do you think?

BTW, if you haven't already, please join our slack, I'd like to have a chat with you and thank you for your contributions :)
https://slack.openzeppelin.org/

**Comment by DavidKnott** [comment]
Thanks for looking it over so quickly @maraoz! I like that idea, I'll integrate it with the ```Pause``` contract and change the names to match the ```PauseToken```.

**Comment by DavidKnott** [comment]
@maraoz I changed the  ```Pausable``` contract like we talked about before.  I think the modifier '''canTransfer''' makes sense for allowing transactions only when they're not paused but ```cannotTransfer``` seems a bit confusing to me (the functionality is that it allows stuff to occur only when paused).  What do  you think?

**Comment by maraoz** [comment]
@DavidKnott I think `Pausable` should have no mentions to transfers, I would call the methods `pause`, `unpause`, and the modifiers `ifPaused` or `whenPaused` and `whenNotPaused`

File: contracts/lifecycle/Pausable.sol [hunk 1]
```diff
@@ -6,34 +6,36 @@ import "../ownership/Ownable.sol";
...
 contract Pausable is Ownable {
-  bool public stopped;
+  event Pause();
+  event Unpause();
 
-  modifier stopInEmergency {
-    if (stopped) {
-      throw;
-    }
+  bool public transfersPaused = false;
```
Comment (maraoz) [inline contracts/lifecycle/Pausable.sol]: `bool public paused = false;`?

File: contracts/lifecycle/Pausable.sol [hunk 2]
```diff
@@ -6,34 +6,36 @@ import "../ownership/Ownable.sol";
...
+
+  modifier canTransfer() {
+    if (transfersPaused) throw;
     _;
   }
```
Comment (maraoz) [inline contracts/lifecycle/Pausable.sol]: whenNotPaused

File: contracts/lifecycle/Pausable.sol [hunk 3]
```diff
@@ -6,34 +6,36 @@ import "../ownership/Ownable.sol";
...
-  
-  modifier onlyInEmergency {
-    if (!stopped) {
-      throw;
-    }
+
+  modifier cannotTransfer {
```
Comment (maraoz) [inline contracts/lifecycle/Pausable.sol]: whenPaused

File: contracts/lifecycle/Pausable.sol [hunk 4]
```diff
@@ -6,34 +6,36 @@ import "../ownership/Ownable.sol";
...
+    if (!transfersPaused) throw;
     _;
   }
 
-  // called by the owner on emergency, triggers stopped state
-  function emergencyStop() external onlyOwner {
-    stopped = true;
+  // called by the owner to pause transfers, triggers stopped state
+  function pauseTransfers() onlyOwner canTransfer returns (bool) {
```
Comment (maraoz) [inline contracts/lifecycle/Pausable.sol]: `pause`

File: contracts/lifecycle/Pausable.sol [hunk 5]
```diff
@@ -6,34 +6,36 @@ import "../ownership/Ownable.sol";
...
+    transfersPaused = true;
+    Pause();
+    return true;
   }
 
-  // called by the owner on end of emergency, returns to normal state
-  function release() external onlyOwner onlyInEmergency {
-    stopped = false;
+  // called by the owner to allow transfers, returns to normal state
+  function unpauseTransfers() onlyOwner cannotTransfer returns (bool) {
```
Comment (maraoz) [inline contracts/lifecycle/Pausable.sol]: unpause

**Comment by DavidKnott** [comment]
@maraoz thanks for the feedback, I took all mentions of transactions out of the ```Pausable``` contract (tried to make it as modular as possible).  Do you see anything else that needs to be changed?

**Comment by maraoz** [comment]
LGTM, many thanks for the contribution!

**Pull request merged**